src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -40,12 +42,14 @@ pub struct Context<'a, 'b: 'a> {
 }
 
 impl<'a, 'b: 'a> Context<'a, 'b> {
-    pub fn new(env: &'a str, resolve: &'a Resolve, sources: &'a SourceMap<'b>,
-               deps: &'a PackageSet, config: &'b Config<'b>,
-               host: Layout, target: Option<Layout>,
-               root_pkg: &Package,
-               build_config: BuildConfig)
-               -> CargoResult<Context<'a, 'b>> {
```

Comment:
This already has an unfortunately large number of arguments, perhaps the execution could be set after a `Context` is created?